hibernate-core/src/main/java/org/hibernate/engine/jdbc/batch/internal/BatchImpl.java

```diff
@@ -302,12 +302,10 @@ protected void performExecution() {
 				}
 				catch (SQLException e) {
 					abortBatch( e );
-					BATCH_MESSAGE_LOGGER.unableToExecuteBatch( e, sql );
 					throw sqlExceptionHelper.convert( e, "could not execute batch", sql );
 				}
 				catch (RuntimeException re) {
 					abortBatch( re );
-					BATCH_MESSAGE_LOGGER.unableToExecuteBatch( re, sql );
 					throw re;
```

Comment:
Should the exception be wrapped somehow so that the SQL is provided as context, similarly to how we handle `SQLException` above?

Otherwise the removal of the log is also removing information (the SQL).

Reply:
Not sure I agree. We're generally trying to move away from the idiom of unnecessary catch-wrap-and-rethrow, since it results in stack traces which are very difficult to understand.

If there's something wrong with the SQL, then we would have got a `JDBCException`. Anything other than a `JDBCException` is much less probable here, and likely doesn't have much to do with the actual SQL we're executing.

Reply:
> Not sure I agree. We're generally trying to move away from the idiom of unnecessary catch-wrap-and-rethrow, since it results in stack traces which are very difficult to understand.

That's my point, it might be necessary to wrap if the context (SQL) is essential to understanding the error.

> If there's something wrong with the SQL, then we would have got a `JDBCException`. Anything other than a `JDBCException` is much less probable here, and likely doesn't have much to do with the actual SQL we're executing.

Fair enough, no context needed then, so no wrapping.

Reply:
Actually, I had a doubt so I did a bit of investigation... the error log with the SQL was introduced precisely because someone needed the SQL to investigate a `RuntimeException`. See: https://github.com/hibernate/hibernate-orm/pull/1544

Reply:
> See: https://github.com/hibernate/hibernate-orm/pull/1544

OK, but we already fixed that problem by correctly reporting a `StaleStateException` in that case.

This log message was the crappy fix to what we've now fixed in a much better way. (As is often the case with log messages.)

Reply:
Sorry, I hadn't seen *all* of you guys' additional comments, should I revert the `BatchFailedException` rethrow back to just a rethrow of the `RuntimException` ?

Reply:
> should I revert the `BatchFailedException` rethrow back to just a rethrow of the `RuntimException`?

Yes.